File: Java/Medium/WrongAnswer/rearrange-array-elements-by-sign_4.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        
        int n = nums.length;
        int[] ans = new int[n];
        int i = 0;
        int j = n - 1;
        Arrays.sort(nums);
        for (int k = 0; k < n; k++) {
            if (k % 2 == 0) {
                ans[k] = nums[i++];
            } else {
                ans[k] = nums[j--];
            }
        }
        return ans;
    }
}

File: Rust/Medium/WrongAnswer/rearrange-array-elements-by-sign_4.rs
impl Solution {
    pub fn rearrange_array(nums: Vec<i32>) -> Vec<i32> {
        
        let mut nums = nums;
        nums.sort_by_key(|&x| x.abs());
        let mut i = 1;
        while i < nums.len() - 1 {
            nums.swap(i, i + 1);
            i += 2;
        }
        nums
    }
}